components/supervisor/pkg/supervisor/config.go

```diff
@@ -468,6 +468,11 @@ func (c WorkspaceConfig) isDebugWorkspace() bool {
 	return c.DebugWorkspaceType != api.DebugWorkspaceType_noDebug
 }
 
+// isImageBuild returns true if the workspace is an image build.
+func (c WorkspaceConfig) isImageBuild() bool {
+	return os.Getenv("BOB_DOCKERFILE_PATH") != ""
```

Comment:
nit: Other config will [read env when init](https://github.com/gitpod-io/gitpod/blob/e266df1bc80b7f887cadb294f402ed6eec37a07b/components/supervisor/pkg/supervisor/config.go#L310), we should do it the same for isImageBuild to align with others

<img width="1101" alt="image" src="https://github.com/user-attachments/assets/54876233-ff15-409f-8863-966ffa11dc1e">

Reply:
Thanks for pointing this out, fixed in [`e1b1d79` (#20281)](https://github.com/gitpod-io/gitpod/pull/20281/commits/e1b1d794305bd66e05a93b2b8c492a069ec68d19).